### PR TITLE
Improve the "certbot certificates" output

### PR DESCRIPTION
**Pull request by pde**
Make `certbot certificates` output slightly more greppable and future-proof:

* easy grepping to see if a cert is valid or not
* a place for future addition of OCSP checks for invalidity due to revocation
* indicate whether a cert is a test / staging cert!
* change the "Valid Until" string to "Expiry Date", because it's confusing if an invalid/expired cert has a  Valid Until" listing, and someone might foolishly grep for "Valid", having never seen what an expired cert  looks like.

Fixes: #3835

### CODE REVIEW

**Comment by pde** [comment]
Before:
```
certbot certificates
Saving debug log to /var/log/letsencrypt/letsencrypt.log

-------------------------------------------------------------------------------
Found the following certs:
  Certificate Name: isnot.org
    Domains: isnot.org goxogle.com is.isnot.org www.isnot.org
    Valid Until: 2017-01-25 22:06:00+00:00 (53 days)
    Certificate Path: /etc/letsencrypt/live/isnot.org/fullchain.pem
    Private Key Path: /etc/letsencrypt/live/isnot.org/privkey.pem
  Certificate Name: goxogle.com
    Domains: goxogle.com www.isnot.org
    Valid Until: 2016-04-07 23:39:00+00:00 (EXPIRED)
    Certificate Path: /etc/letsencrypt/live/goxogle.com/fullchain.pem
    Private Key Path: /etc/letsencrypt/live/goxogle.com/privkey.pem
  Certificate Name: demo.reworld.org
    Domains: demo.reworld.org
    Valid Until: 2017-01-02 21:23:00+00:00 (30 days)
    Certificate Path: /etc/letsencrypt/live/demo.reworld.org/fullchain.pem
    Private Key Path: /etc/letsencrypt/live/demo.reworld.org/privkey.pem
  Certificate Name: www.isnot.org
    Domains: www.isnot.org
    Valid Until: 2017-02-28 23:41:00+00:00 (87 days)
    Certificate Path: /etc/letsencrypt/live/www.isnot.org/fullchain.pem
    Private Key Path: /etc/letsencrypt/live/www.isnot.org/privkey.pem
  Certificate Name: isnot.org-0001
    Domains: isnot.org
    Valid Until: 2017-01-02 21:23:00+00:00 (30 days)
    Certificate Path: /etc/letsencrypt/live/isnot.org-0001/fullchain.pem
    Private Key Path: /etc/letsencrypt/live/isnot.org-0001/privkey.pem
-------------------------------------------------------------------------------
```

After:
```
certbot certificates
Saving debug log to /var/log/letsencrypt/letsencrypt.log

-------------------------------------------------------------------------------
Found the following certs:
  Certificate Name: isnot.org
    Domains: isnot.org goxogle.com is.isnot.org www.isnot.org
    Expiry Date: 2017-01-25 22:06:00+00:00 (VALID: 53 days)
    Certificate Path: /etc/letsencrypt/live/isnot.org/fullchain.pem
    Private Key Path: /etc/letsencrypt/live/isnot.org/privkey.pem
  Certificate Name: goxogle.com
    Domains: goxogle.com www.isnot.org
    Expiry Date: 2016-04-07 23:39:00+00:00 (INVALID: EXPIRED)
    Certificate Path: /etc/letsencrypt/live/goxogle.com/fullchain.pem
    Private Key Path: /etc/letsencrypt/live/goxogle.com/privkey.pem
  Certificate Name: demo.reworld.org
    Domains: demo.reworld.org
    Expiry Date: 2017-01-02 21:23:00+00:00 (VALID: 30 days)
    Certificate Path: /etc/letsencrypt/live/demo.reworld.org/fullchain.pem
    Private Key Path: /etc/letsencrypt/live/demo.reworld.org/privkey.pem
  Certificate Name: www.isnot.org
    Domains: www.isnot.org
    Expiry Date: 2017-02-28 23:41:00+00:00 (INVALID: TEST CERT)
    Certificate Path: /etc/letsencrypt/live/www.isnot.org/fullchain.pem
    Private Key Path: /etc/letsencrypt/live/www.isnot.org/privkey.pem
  Certificate Name: isnot.org-0001
    Domains: isnot.org
    Expiry Date: 2017-01-02 21:23:00+00:00 (VALID: 30 days)
    Certificate Path: /etc/letsencrypt/live/isnot.org-0001/fullchain.pem
    Private Key Path: /etc/letsencrypt/live/isnot.org-0001/privkey.pem
-------------------------------------------------------------------------------
```

File: certbot/cert_manager.py [hunk 1]
```diff
@@ -39,20 +39,25 @@ def _report_human_readable(parsed_certs):
     certinfo = []
     for cert in parsed_certs:
         now = pytz.UTC.fromutc(datetime.datetime.utcnow())
-        if cert.target_expiry <= now:
-            expiration_text = "EXPIRED"
+        # TODO: perhaps one day we should actually ask OpenSSL to double check whether the
+        # certificate appears to be valid here, but that would require a dependency on
+        # either the openssl binary or PyOpenSSL 0.15 https://eff.org/r.pyopenssl-verify
+        if cert.is_test_cert:
+            expiration_text = "INVALID: TEST CERT"
```
Comment (ohemorange) [inline certbot/cert_manager.py]: Yes woohoo I love this!

File: certbot/cert_manager.py [hunk 2]
```diff
@@ -39,20 +39,25 @@ def _report_human_readable(parsed_certs):
     certinfo = []
     for cert in parsed_certs:
         now = pytz.UTC.fromutc(datetime.datetime.utcnow())
-        if cert.target_expiry <= now:
-            expiration_text = "EXPIRED"
+        # TODO: perhaps one day we should actually ask OpenSSL to double check whether the
```
Comment (ohemorange) [inline certbot/cert_manager.py]: Could you put this in an issue instead of a TODO?

File: certbot/storage.py [hunk 3]
```diff
@@ -281,6 +282,15 @@ def archive_dir(self):
             return os.path.join(
                 self.cli_config.default_archive_dir, self.lineagename)
 
+    @property
+    def is_test_cert(self):
+        """Returns true if this is a test cert from a staging server."""
+        server = self.configuration["renewalparams"].get("server", None)
+        if server:
```
Comment (ohemorange) [inline certbot/storage.py]: When would server be None?

File: certbot/storage.py [hunk 4]
```diff
@@ -281,6 +282,15 @@ def archive_dir(self):
             return os.path.join(
                 self.cli_config.default_archive_dir, self.lineagename)
 
+    @property
+    def is_test_cert(self):
```
Comment (ohemorange) [inline certbot/storage.py]: This method doesn't have a unit test, could you add one?

File: certbot/storage.py (hunk 3 above)
Comment (pde) [inline certbot/storage.py]: That's the common case. If it's left as the default value, it won't be saved to the renewal conf file, so it won't be in the `renewalparams` object that's read back, so the `.get()` method above will set it to none.

File: certbot/cert_manager.py (hunk 2 above)
Comment (pde) [inline certbot/cert_manager.py]: Moved to #3866

**Pull request merged**